### DISCUSSION TITLE
Issues building macOS binaries

### DISCUSSION THREAD

**Discussion by wintermute001** [Xpra-org/xpra#4018] [Q&A]
On an M2 Macbook pro with Sonoma (macOS 14.0), "jhbuild update" fails multiple times

examples:
1. `*** Error during phase force_checkout of zlib: file hash is incorrect (expected d14c38e313afc35a9a8760dadf26042f51ea0f5d154b0630a31da0540107fb98, got d5a89e26beae0bc03ad18a0b0d1d3d75f87c32047879d25da11970cb5c4662a3) *** [4/137]`
 
2. `*** Error during phase build of pygments: [Errno 2] No such file or directory: '/Users/fablap01/gtk/inst/bin/python3' *** [9/137]`
3. `clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [libffi.la] Error 1
make[1]: *** [all-recursive] Error 1
make: *** [all] Error 2
*** Error during phase build of libffi: ########## Error running make -j 11  *** [12/137]`

What is the recommendation? Ignoring all the errors and continuing the installation?

**Comment by totaam** [comment]
* `zlib`
I don't think you can continue without zlib.
Figure out if the checksum is wrong, and if so, change it - then ideally submit a PR.

* `pygments` I don't remember what I did with this one, IIRC it's only used for building other stuff so you could just `pip install` it from a jhbuild shell and ignore the module?

* `libffi` - no details included so no idea, sorry

Edit: originated from https://github.com/Xpra-org/xpra/issues/4017

**Reply by wintermute001** [reply]
Thanks. Will check zlib in more details; some of the issues is that the tool seems to be looking for python3 in the wrong location (I have it on a homebrew local folder....): do you know where I can tweak the jhbuild search path?
Thanks again

**Reply by totaam** [reply]
Sorry, no idea.

**Reply by totaam** [reply]
For `libffi`, please see:
https://github.com/Xpra-org/gtk-osx-build/issues/22
https://github.com/Xpra-org/gtk-osx-build/issues/27

**Comment by totaam** [comment]
There have been many fixes to the modulesets so the issues above are likely outdated - except for `libffi`.

Please open a new ticket if you are still having issues.
